storage/Devices/LuksImpl.cc

```diff
@@ -400,7 +419,31 @@ namespace storage
 	     */
 	    string dm_table_name;
 	    if (it2 != cmd_dmsetup_table.end())
+	    {
 		dm_table_name = it2->first;
+
+		/*
+		 * If integrity is set the name will ends with _dif,
+		 * and later we will find the target with the correct
+		 * name
+		 */
+		if (boost::ends_with(dm_table_name, "_dif")) {
+		    bool integrity = false;
+		    for (const CmdDmsetupTable::Table& table : it2->second)
+		    {
+			if (table.target == "integrity")
+			    integrity = true;
+		    }
+
+		    if (integrity)
+		    {
+			dm_table_name = dm_table_name.substr(0, dm_table_name.length() - 4);
+			const CmdDmsetupInfo& cmd_dmsetup_info = system_info.getCmdDmsetupInfo();
+			if (!cmd_dmsetup_info.exists(dm_table_name))
+			    ST_THROW(Exception("integrity device found, missing the associated 'crytp' target"));
```

Comment:
I am just talking about the typo here: "crytp" -> "crypt"

Reply:
I am blind. Done. (Sorry)